nsqd/buffer_pool.go

```diff
@@ -18,5 +18,6 @@ func bufferPoolGet() *bytes.Buffer {
 }
 
 func bufferPoolPut(b *bytes.Buffer) {
+	b.Reset()
```

Comment:
This is interesting, what does this imply was happening before without this? Were the pooled buffers _growing_ because they had not been reset?

Reply:
i find that the `writeMessageToBackend` of `message.go` also called buf.Reset, I think it should be called in bufferPoolPut

Reply:
There were a couple different schemes for caching the `bytes.Buffer` for `writeMessageToBackend()`, so the reset was in the latter function ... it all worked, and still does. I think the remaining task is to make it so there aren't redundant `Reset()` calls, which are a bit confusing. I think there are two options:
   * take it out of `writeMessageToBackend()`, and then put it in the callers that don't use the buffer pool (they re-use one buffer for a loop of message ops, which is probably more efficient than going through the pool)
   * change `writeMessageToBackend()` to call `bufferPoolGet()` itself, and remove the `*bytes.Buffer` param to it

Reply:
done, pls check again

Reply:
looks good to me ... can you repeat the benchmark once more please?